### PR TITLE
APEXMALHAR-2409 PojoInnerJoin accumulation to emit a POJO

### PR DESCRIPTION
**Pull request by chinmaykolhatkar**
@tushargosavi Please review and merge.

### CODE REVIEW

File: library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java [hunk 1]
```diff
@@ -132,14 +134,36 @@ public PojoInnerJoin(int num, String... keys)
...
+        throw Throwables.propagate(e);
+      }
+
+      for (Map.Entry<String, Object> entry : resultMap.entrySet()) {
+        try {
+          Field f = outClass.getDeclaredField(entry.getKey());
+          f.setAccessible(true);
+          f.set(o, entry.getValue());
+        } catch (Throwable e) {
+          e.printStackTrace();
```
Comment (tushargosavi) [inline library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java]: avoid printStackTrace, rethrow instead.

File: library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java [hunk 2]
```diff
@@ -36,22 +36,24 @@
  * @since 3.6.0
  */
 public class PojoInnerJoin<InputT1, InputT2>
-    implements MergeAccumulation<InputT1, InputT2, List<List<Map<String, Object>>>, List<Map<String, Object>>>
+    implements MergeAccumulation<InputT1, InputT2, List<List<Map<String, Object>>>, List<?>>
 {
   protected String[] keys;
```
Comment (tushargosavi) [inline library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java]: make this final

File: library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java [hunk 3]
```diff
@@ -36,22 +36,24 @@
...
+  protected Class<?> outClass;
```
Comment (tushargosavi) [inline library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java]: make it final

File: library/src/test/java/org/apache/apex/malhar/lib/window/impl/PojoInnerJoinTestApplication.java [hunk 4]
```diff
@@ -375,7 +375,7 @@ public void populateDAG(DAG dag, Configuration conf)
     productGenerator.setSalesEvent(false);
     WindowedMergeOperatorImpl<POJOGenerator.SalesEvent, POJOGenerator.ProductEvent, List<Set<Object>>, List<List<Object>>> op
         = dag.addOperator("Merge", new WindowedMergeOperatorImpl<POJOGenerator.SalesEvent, POJOGenerator.ProductEvent, List<Set<Object>>, List<List<Object>>>());
-    op.setAccumulation(new PojoInnerJoin(2,"productId","productId"));
+    op.setAccumulation(new PojoInnerJoin(2, Object.class, "productId","productId"));
```
Comment (tushargosavi) [inline library/src/test/java/org/apache/apex/malhar/lib/window/impl/PojoInnerJoinTestApplication.java:378]: Have you run this app? setting fields on Object will fail in getOutput

**Comment by tushargosavi** [comment]
Fix following checkstyle error. 
src/test/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoinTest.java:[170,47] (whitespace) NoWhitespaceAfter: ')' is followed by whitespace.

File: library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java (hunk 2 above)
Comment (chinmaykolhatkar) [reply library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java]: Done.

File: library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java (hunk 3 above)
Comment (chinmaykolhatkar) [reply library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java]: Done.

File: library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java (hunk 1 above)
Comment (chinmaykolhatkar) [reply library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/PojoInnerJoin.java]: Done.

File: library/src/test/java/org/apache/apex/malhar/lib/window/impl/PojoInnerJoinTestApplication.java (hunk 4 above)
Comment (chinmaykolhatkar) [reply library/src/test/java/org/apache/apex/malhar/lib/window/impl/PojoInnerJoinTestApplication.java:378]: Yes it runs.

**Comment by chinmaykolhatkar** [comment]
@tushargosavi Done the changes

**Pull request merged**